cmd/porter/version.go

```diff
@@ -7,15 +7,18 @@ import (
 )
 
 func buildVersionCommand(p *porter.Porter) *cobra.Command {
-	opts := version.Options{}
+	opts := version.VersionOpts{}
 	cmd := &cobra.Command{
 		Use:   "version",
 		Short: "Print the application version",
 		PreRunE: func(cmd *cobra.Command, args []string) error {
-			return opts.Validate()
+			return opts.Options.Validate()
 		},
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return p.PrintVersion(opts)
+			if opts.System {
```

Comment:
The convention in Porter is that all logic should be in `pkg/porter`, and `cmd/porter` should just wire up cobra to functions defined in that package. 

Please move the check against `opts.System` into `PrintVersion`, so that the code in this file goes back to looking like:

```go
RunE: func(cmd *cobra.Command, args []string) error {
   return p.PrintVersion(opts)
}
```

If you are curious about why, I did a [talk about how Porter is designed at GopherCon](https://youtu.be/eMz0vni6PAw). The short answer is that it helps us test, and allows for people to build on top of porter's libraries.